addons/l10n_id_efaktur/tests/test_l10n_id_efaktur.py

```diff
@@ -88,3 +89,63 @@ def test_efaktur_csv_output_decimal_place(self):
         self.assertEqual(efaktur_csv_expected, efaktur_csv_output)
+
+    def test_efaktur_total_rounding_accuracy(self):
+        """ Use case:
+        Using a tax of 11% price included on every line:
+
+        line | qty | price | subtotal
+        1    | 24  | 57851 | 1250832.43
+        2    | 24  | 65184 | 1409383.78
+        3    | 24  | 77134 | 1667762.16
+        4    | 24  | 87835 | 1899135.14
+        5    | 20  | 180342| 3249405.41
+
+        Untaxed Amount: 9474250.92
+        Taxes: 1042176.08
+        Total: 10518936.00
+
+        Efaktur will display both:
+        -The detail of the lines.
+        and
+        - the amount_untaxed and amount_tax rounded to 0 decimals.
+
+        The sum of the lines MUST exactly match with the amount_untaxed and amount_tax.
+        Which most of the case won't be happening because sum(rounded(vals)) is usually not equal to rounded(sum(vals))
+        when using integers.
+
+        To remediate to that issue we are putting the difference in the amount of the first line.
+        """
+        # Prepare the test invoice.
+        tax_id = self.env["account.tax"].create(
+            {"name": "test tax 11", "type_tax_use": "sale", "amount": 11.0, "price_include": True}
+        )
+        invoice = self.env['account.move'].create({
+            'move_type': 'out_invoice',
+            'partner_id': self.partner_id.id,
+            'invoice_date': '2019-05-01',
+            'date': '2019-05-01',
+            'invoice_line_ids': [
+                (0, 0, {'name': 'line1', 'price_unit': 57851, 'quantity': 24, 'tax_ids': tax_id.ids}),
+                (0, 0, {'name': 'line2', 'price_unit': 65184, 'quantity': 24, 'tax_ids': tax_id.ids}),
+                (0, 0, {'name': 'line3', 'price_unit': 77134, 'quantity': 24, 'tax_ids': tax_id.ids}),
+                (0, 0, {'name': 'line4', 'price_unit': 87835, 'quantity': 24, 'tax_ids': tax_id.ids}),
+                (0, 0, {'name': 'line5', 'price_unit': 180342, 'quantity': 20, 'tax_ids': tax_id.ids}),
+            ],
+            'l10n_id_kode_transaksi': '01'
+        })
+        invoice.action_post()
+        # Generate the generate efaktur csv.
+        efaktur_csv_output = invoice._generate_efaktur_invoice(',')
+        # Validate the result: the sum of the lines must exactly match with the amount_untaxed and amount_tax.
+        split_file = efaktur_csv_output.split('\n')
+        total_lines = split_file[3].split(',')
+        amount_untaxed_total = int(total_lines[10].replace('"', ''))
+        amount_tax_total = int(total_lines[11].replace('"', ''))
+
+        lines = split_file[4:9]
+        amount_untaxed_sum = sum([float(line.split(',')[7].replace('"', '')) for line in lines])
+        amount_tax_sum = sum([float(line.split(',')[8].replace('"', '')) for line in lines])
+
+        self.assertEqual(amount_untaxed_total, amount_untaxed_sum)
+        self.assertEqual(amount_tax_total, amount_tax_sum)
```

Comment:
This part is very cryptic to me with all those `[number]`

Maybe that could be a solution to help making this smaller/more readable?
```py
import csv

dict_reader = csv.DictReader(efaktur_csv_output.splitlines(), delimiter=',')
amount_untaxed_sum = sum(row["the_correct_untaxed_column_name"] for row in dict_reader)
amount_tax_sum = sum(row["the_correct_tax_column_name"] for row in dict_reader)
```

Reply:
Thanks for the advice, DictReader looks great for this :+1: 
I couldn't really do it the way you proposed as the csv as a particular structure, not just one title row and then data,
but with that I could do a loop and get a clearer/more readable code.